File: app/src/main/java/fr/free/nrw/commons/utils/AbstractTextWatcher.java
package fr.free.nrw.commons.utils;

import android.text.Editable;
import android.text.TextWatcher;

import androidx.annotation.NonNull;

public class AbstractTextWatcher implements TextWatcher {
    private final TextChange textChange;

    public AbstractTextWatcher(@NonNull TextChange textChange) {
        this.textChange = textChange;
    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {
        textChange.onTextChanged(s.toString());
    }

    @Override
    public void afterTextChanged(Editable s) {
    }

    public interface TextChange {
        void onTextChanged(String value);
    }
}

File: app/src/main/java/fr/free/nrw/commons/utils/AbstractTextWatcher.kt
package fr.free.nrw.commons.utils

import android.text.Editable
import android.text.TextWatcher

class AbstractTextWatcher(
    private val textChange: TextChange
) : TextWatcher {

    override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
        // No-op
    }

    override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
        textChange.onTextChanged(s.toString())
    }

    override fun afterTextChanged(s: Editable?) {
        // No-op
    }

    interface TextChange {
        fun onTextChanged(value: String)
    }
}
